packages/agent/src/app.ts

```diff
@@ -156,12 +157,38 @@ export class App {
           // @ts-expect-error - Deprecated message type
           case 'transmit':
           case 'agent:transmit:response':
-            this.addToHl7Queue(command);
+            if (this.config?.status !== 'active') {
+              const err = new Error('Agent.status is currently set to off');
+              this.log.error(normalizeErrorString(err));
+              this.addToWebSocketQueue({
+                type: 'agent:transmit:response',
+                channel: command.channel,
+                remote: command.remote,
+                callback: command.callback,
+                contentType: ContentType.TEXT,
+                statusCode: 400,
+                body: normalizeErrorString(err),
+              } satisfies AgentTransmitResponse);
+            } else {
+              this.addToHl7Queue(command);
+            }
             break;
           // @ts-expect-error - Deprecated message type
           case 'push':
           case 'agent:transmit:request':
-            if (command.contentType === ContentType.PING) {
+            if (this.config?.status !== 'active') {
+              const err = new Error('Agent.status is currently set to off');
```

Comment:
Given similarity to `case 'agent:transmit:response`, would it be possible to move this to helper function?

Reply:
Done 👍